framework/event_system/event.h

```diff
@@ -10,18 +10,49 @@ namespace chi
 class Event
 {
 public:
-  Event(const std::string& name, int code);
-  Event(const std::string& name, int code, const ParameterBlock& parameter_block);
+  enum ECode
```

Comment:
Particularly since this is public, I would consider naming this `EventCode` or just `Code`.

Reply:
I started with `Code`, but then there is a `Code()` method, so I think `EventCode` will be better...